test/test_tensor.h

```diff
@@ -926,14 +926,25 @@ TEST(tensor, nd1) {
 
 }
 
-TEST(tensor, print) {
-    Tensor<float_t, 4> t({3,2,2,2});
+template<size_t T> std::ostream& print_tester(std::ostream &os) {
+    os << "\nPrinting Tensor of size " << T << ":\n\n";
+    std::vector<size_t> tmp(T, 2);
```

Comment:
Local variable name `tmp` makes the code ambiguous.
What about renaming it to `shape`?